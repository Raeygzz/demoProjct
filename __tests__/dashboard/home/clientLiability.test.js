import React from 'react';
import 'react-native';

import ClientLiability from '../../../src/screens/dashboard/home/clientLiability';

import clientIssueArr from '../../../src/helper/constants/constants';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


import sinon from 'sinon';


import Enzyme from 'enzyme';
import { shallow, mount, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


describe('<ClientLiability />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(<ClientLiability />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders View, Text, StyleSheet, ScrollView, RupiSelect, NormalButton, InvertButton, RupiTextInput & clientIssueArr components using (enzyme)shallow,find (chai)expect,to.have,lengthOf', () => {
    const wrapper = shallow(<ClientLiability />);
    expect(wrapper.find('View').exists()).toEqual(true);
    expect(wrapper.find('Text').exists()).toEqual(true);
    expect(wrapper.find('StyleSheet')).toBeDefined();
    expect(wrapper.find('ScrollView')).toBeTruthy();
    expect(wrapper.find('RupiSelect').exists()).toEqual(true);
    expect(wrapper.find('InvertButton').exists()).toEqual(true);
    expect(wrapper.find('NormalButton').exists()).toEqual(true);
    expect(wrapper.find('RupiTextInput').exists()).toEqual(true);
    expect(wrapper.find('clientIssueArr').exists());
    expect(wrapper.find('clientIssueArr')).toBeTruthy();
  });


  it('test state and state value using (enzyme)shallow,state,instance,expect & (jest)toEqual,toBeTruthy,toBeDefined,toBe', () => {
    const wrapper = shallow(<ClientLiability />);
    expect(wrapper.state()).toBeTruthy();
    expect(wrapper.state()).toBeDefined();
    expect(wrapper.state('clientIssue')).toBe('');
    expect(wrapper.state('clientIssueOptions')).toEqual([
      { id: 1, value: "Money Pending To Receive", label: "Money Pending To Receive" },
      { id: 2, value: "Money Pending To Pay", label: "Money Pending To Pay" },
      { id: 3, value: "Last Deal With Client", label: "Last Deal With Client" },
      { id: 4, value: "Client Not Applicable For Scheme", label: "Client Not Applicable For Scheme" },
      { id: 5, value: "Client Payment Delay More Than 3 Months", label: "Client Payment Delay More Than 3 Months" },
    ]);
    expect(wrapper.state('clientNameOptions')).toEqual([
      { id: 1, value: 'Regan Timsina', label: 'Regan Timsina' },
      { id: 2, value: 'Gyanu Ojha', label: 'Gyanu Ojha' },
      { id: 3, value: 'Keshav Gautam', label: 'Keshav Gautam' },
      { id: 4, value: 'Romisha Timsina', label: 'Romisha Timsina' },
      { id: 5, value: 'Pukar Ojha', label: 'Pukar Ojha' }
    ]);
  })

})
