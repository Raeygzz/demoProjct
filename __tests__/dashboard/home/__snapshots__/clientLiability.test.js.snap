// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientLiability /> renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 20,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "white",
              "borderBottomWidth": 1,
              "height": 68,
            },
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "marginBottom": 20,
              "minHeight": 68,
              "overflow": "hidden",
              "paddingTop": 20,
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#71648E",
                  "fontSize": 12,
                },
                Object {
                  "position": "absolute",
                  "right": 18,
                  "top": 18,
                },
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Apercu_Regular",
                "fontSize": 14,
                "left": 0,
                "lineHeight": 17,
                "opacity": 0.68,
                "position": "absolute",
                "top": 16,
              }
            }
          >
            Client Name
          </Text>
          <Text
            minimumFontScale={16}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "bottom": 16,
                    "fontSize": 18,
                    "left": 0,
                    "position": "absolute",
                  },
                  Object {
                    "fontFamily": "Apercu-Medium",
                  },
                ],
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": 50,
                  "margin": 0,
                  "marginLeft": -9,
                  "opacity": 0,
                  "padding": 0,
                },
                Object {
                  "fontFamily": "Apercu-Bold",
                },
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "fontFamily": "Apercu Medium",
                  "fontSize": 18,
                  "padding": 0,
                },
                null,
                Object {
                  "color": "white",
                },
              ]
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Select Client Name",
                    "textColor": undefined,
                    "value": null,
                  },
                  Object {
                    "label": "Regan Timsina",
                    "textColor": undefined,
                    "value": "Regan Timsina",
                  },
                  Object {
                    "label": "Gyanu Ojha",
                    "textColor": undefined,
                    "value": "Gyanu Ojha",
                  },
                  Object {
                    "label": "Keshav Gautam",
                    "textColor": undefined,
                    "value": "Keshav Gautam",
                  },
                  Object {
                    "label": "Romisha Timsina",
                    "textColor": undefined,
                    "value": "Romisha Timsina",
                  },
                  Object {
                    "label": "Pukar Ojha",
                    "textColor": undefined,
                    "value": "Pukar Ojha",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "white",
              "borderBottomWidth": 1,
              "height": 68,
            },
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "marginBottom": 20,
              "minHeight": 68,
              "overflow": "hidden",
              "paddingTop": 20,
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#71648E",
                  "fontSize": 12,
                },
                Object {
                  "position": "absolute",
                  "right": 18,
                  "top": 18,
                },
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Apercu_Regular",
                "fontSize": 14,
                "left": 0,
                "lineHeight": 17,
                "opacity": 0.68,
                "position": "absolute",
                "top": 16,
              }
            }
          >
            Client Issue
          </Text>
          <Text
            minimumFontScale={16}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "bottom": 16,
                    "fontSize": 18,
                    "left": 0,
                    "position": "absolute",
                  },
                  Object {
                    "fontFamily": "Apercu-Medium",
                  },
                ],
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": 50,
                  "margin": 0,
                  "marginLeft": -9,
                  "opacity": 0,
                  "padding": 0,
                },
                Object {
                  "fontFamily": "Apercu-Bold",
                },
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "fontFamily": "Apercu Medium",
                  "fontSize": 18,
                  "padding": 0,
                },
                null,
                Object {
                  "color": "white",
                },
              ]
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Select Client Issue",
                    "textColor": undefined,
                    "value": null,
                  },
                  Object {
                    "label": "Money Pending To Receive",
                    "textColor": undefined,
                    "value": "Money Pending To Receive",
                  },
                  Object {
                    "label": "Money Pending To Pay",
                    "textColor": undefined,
                    "value": "Money Pending To Pay",
                  },
                  Object {
                    "label": "Last Deal With Client",
                    "textColor": undefined,
                    "value": "Last Deal With Client",
                  },
                  Object {
                    "label": "Client Not Applicable For Scheme",
                    "textColor": undefined,
                    "value": "Client Not Applicable For Scheme",
                  },
                  Object {
                    "label": "Client Payment Delay More Than 3 Months",
                    "textColor": undefined,
                    "value": "Client Payment Delay More Than 3 Months",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "marginBottom": 20,
              "minHeight": 68,
              "overflow": "hidden",
              "paddingTop": 20,
              "position": "relative",
            },
            undefined,
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Apercu_Regular",
              "fontSize": 14,
              "left": 0,
              "lineHeight": 17,
              "opacity": 0.68,
              "position": "absolute",
              "top": 26,
            }
          }
        >
          Client Issue Description
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          multiline={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={null}
          placeholderTextColor="rgba(255,255,255,0.7)"
          rejectResponderTermination={true}
          returnKeyType="done"
          selectionColor="#71648E"
          style={
            Array [
              Object {
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "color": "white",
                "fontFamily": "Apercu Medium",
                "fontSize": 18,
                "padding": 0,
              },
              null,
              undefined,
              Object {
                "borderBottomColor": "white",
                "height": 37,
                "lineHeight": 12,
              },
              null,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 30,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffbcbc",
                "borderColor": "red",
                "borderWidth": 1,
                "height": 100,
                "justifyContent": "center",
                "margin": 10,
                "width": 300,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "margin": 20,
              }
            }
          >
            TouchableNativeFeedback is not supported on this platform!
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffbcbc",
                "borderColor": "red",
                "borderWidth": 1,
                "height": 100,
                "justifyContent": "center",
                "margin": 10,
                "width": 300,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "margin": 20,
              }
            }
          >
            TouchableNativeFeedback is not supported on this platform!
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "red",
            "marginTop": 30,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Regan Timsina
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Money Pending To Pay :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "green",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Gyanu Ojha
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Money Pending To Receive :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "lightgreen",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Keshav Gautam
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Last Deal With Client :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "pink",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Romisha Timsina
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Client Not Applicable For Scheme :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "darkgrey",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Pukar Ojha
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Client Payment Delay More Than 3 Months :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
